metadata-ingestion/src/datahub/api/entities/assertion/compiler_interface.py

```diff
@@ -0,0 +1,63 @@
+from abc import abstractmethod
+from dataclasses import dataclass, field
+from enum import Enum
+from pathlib import Path
+from typing import Dict, List
+
+from datahub.api.entities.assertion.assertion_config_spec import AssertionsConfigSpec
+from datahub.ingestion.api.report import Report
+from datahub.utilities.lossy_collections import LossyDict, LossyList
+
+
+class StrEnum(str, Enum):
+    pass
+
+
+class CompileResultArtifactType(StrEnum):
+    SQL_QUERIES = "SQL_QUERIES"
+    COMPILE_REPORT = "COMPILE_REPORT"
+
+
+@dataclass
+class CompileResultArtifact(Report):
+    name: str
+    type: CompileResultArtifactType
+    description: str
+    path: Path
+
+
+@dataclass
+class AssertionCompilationReport(Report):
```

Comment:
Can we call this AssertionCompilationResult?

And result can have a nested Report object.

It makes sense to call

compiler.compile() -> and get back an AssertionCompilationResult